miner/miner.go

```diff
@@ -0,0 +1,94 @@
+package miner
```

Comment:
Can we introduce an interface for `MiningHeader` or maybe `Miner`? I can imagine a scenario where we want to do an integration test where we want to mine a block (without waiting).